app/models/cohort_list.rb

```diff
@@ -48,6 +51,16 @@ def parse_rows!
         .map { CohortListRow.new(_1) }
   end
 
+  def generate_patients!
+    return if invalid?
```

Comment:
With the exclamation in the method name, I kind of expect these to raise an error. I know that's not the only meaning for `!` in methods, but it would be pretty easy to just use `validate!` here, or would break things?

Reply:
Then we'd have to catch them... Isn't that unusual? I used `!` here to signify that it's a method that mutates the instance.

Reply:
In Ruby `!` is used for variants of methods that may do unexpected things. I don't think there are many, if any, examples of methods that only have a `!` version.

Here the method isn't doing anything unexpected, so not sure it needs the `!`, but it's a minor issue.